lib/manageiq/util/file_splitter.rb

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env ruby
+#
+# Yes, you guessed it... this is basically the `split` command...
+#
+# The intent of this is to allow for greater flexibility and utility than is
+# provided by the BSD/GNU variants however, specifically allowing to pipe from
+# a pg_dump directly to a upload target (whether it be a mounted volume or
+# something like a FTP endpoint, though the former is basically already
+# supported with vanilla `split`).
+#
+# FTP, specifically, will be supported natively since ruby has Net::FTP support
+# built in (feature WIP).  This also allows for this to work correctly cross
+# platform, without having to be concerned with differences in `split`
+# functionality.
+
+require_relative 'ftp_lib'
+require 'optparse'
+
+module ManageIQ
+  module Util
+    class FileSplitter
+      include ManageIQ::Util::FtpLib
+
+      KILOBYTE = 1024
+      MEGABYTE = KILOBYTE * 1024
+      GIGABYTE = MEGABYTE * 1024
+
+      BYTE_HASH = {
+        "k" => KILOBYTE,
+        "m" => MEGABYTE,
+        "g" => GIGABYTE
+      }.freeze
```

Comment:
This addresses @Fryguy 's [comment](https://github.com/ManageIQ/manageiq/pull/17652#discussion_r199016038) in https://github.com/ManageIQ/manageiq/pull/17652

Not using `activesupport` because I might go as far as disabling `rubygems` as well to decrease the boottime of this script, since this script will definitely be the bottleneck when run with piped output from either `pg_dump` or `pg_basebackup`.